fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingBase.java

```diff
@@ -518,33 +518,31 @@ private CompletableFuture<Void> setScrubberTypeOrThrow(FDBRecordStore store) {
                 "Not a scrubber type-stamp");
 
         final Index index = common.getIndex(); // Note: the scrubbers do not support multi target (yet)
-        final Subspace indexesRangeSubspace = indexScrubIndexRangeSubspace(store, index);
-        final Subspace recordsRangeSubspace = indexScrubRecordsRangeSubspace(store, index);
-        RangeSet indexRangeSet = new RangeSet(indexesRangeSubspace);
-        RangeSet recordsRangeSet = new RangeSet(recordsRangeSubspace);
-        AsyncIterator<Range> indexRanges = indexRangeSet.missingRanges(tr).iterator();
-        AsyncIterator<Range> recordsRanges = recordsRangeSet.missingRanges(tr).iterator();
-        return indexRanges.onHasNext().thenCompose(hasNextIndex -> {
-            if (Boolean.FALSE.equals(hasNextIndex)) {
+        IndexingRangeSet indexRangeSet = IndexingRangeSet.forScrubbingIndex(store, index);
+        IndexingRangeSet recordsRangeSet = IndexingRangeSet.forScrubbingRecords(store, index);
+        final CompletableFuture<Range> indexRangeFuture = indexRangeSet.firstMissingRangeAsync();
+        final CompletableFuture<Range> recordRangeFuture = recordsRangeSet.firstMissingRangeAsync();
+        return indexRangeFuture.thenCompose(indexRange -> {
+            if (indexRange == null) {
                 // Here: no un-scrubbed index range was left for this call. We will
                 // erase the 'ranges' data to allow a fresh index re-scrubbing.
                 if (LOGGER.isInfoEnabled()) {
                     LOGGER.info(KeyValueLogMessage.build("Reset scrubber's index range")
                             .addKeysAndValues(common.indexLogMessageKeyValues())
                             .toString());
                 }
-                tr.clear(indexesRangeSubspace.range());
+                tr.clear(indexScrubIndexRangeSubspace(store, index).range());
```

Comment:
Sure, I think adding a `clear` method is a good idea. I've added one, and I've changed explicit clears to call the new method. I can't make the subspace methods private (or package-private, even) though because the `IndexingBase` class and new `IndexingRangeSet` are currently in different packages. If we moved `IndexingBase` (which I think is the idea, though probably in a separate PR to avoid mixing too many refactoring together, as @ScottDugas suggests), we could change that